File: C++ Codes/madras_library.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
using namespace std;
int binaryToDecimal(int n);
int main(){
   fstream newfile;
   char name[25];
   cout <<"Enter file name with extension = \n";
   scanf("%s", name);
   newfile.open(name,ios::in); 
   string st1 = "";
   string decode = "";
   if (newfile.is_open()){   
      string tp;
      while(getline(newfile, tp))
	  { 
          for(int i=0;i<tp.length();i++)
          {
          	if(tp[i] == '*')
          	{
          		st1+='1';
			  }
			else
			{
				st1+='0';
			}
		  }
		stringstream ob(st1); 
    	int x = 0; 
    	ob >> x; 
		int b = binaryToDecimal(x);
		decode = decode + (char)b;
		st1 ="";  
         
      }
      cout << "Decoded Text =  " << decode;
      newfile.close(); 
   }
}
int binaryToDecimal(int n) 
{ 
    int num = n; 
    int dec = 0; 
    int base = 1;  
    int temp = num; 
    
    while (temp) { 
        int last_digit = temp % 10; 
        temp = temp / 10;   
        dec += last_digit * base;   
        base = base * 2; 
    } 
  
    return dec; 
} 

File: Python Codes/Madras_library.py

def binaryToDecimal(n): 
    num = n
    dec = 0
    base = 1       
    temp = num
    while(temp): 
        last_digit = temp % 10
        temp = int(temp / 10)          
        dec += last_digit * base
        base = base * 2
    return dec; 
def decipher():
    s = input("Enter the file name without extension : ")
    f = open(s+".txt", "r")
    decode = ""
    for i in f:
        st = list(i)
        st1 = ""
        for j in range(0,len(st)-1):
            if(st[j] == '*'):
                st1+='1'
            else:
                st1+='0'
        
        b = binaryToDecimal(int(st1)) 
        decode = decode + chr(b)
    return decode
if __name__ == "__main__":   
    
    print("Your decoded text is : \n", decipher())


